### PR TITLE
bootterm: init at 0.5

### PR DESCRIPTION
**Pull request by deadbaed**
I started to use nix recently, and wanted to use [bootterm](https://github.com/wtarreau/bootterm) when using one of my SBCs. I noticed it was not packaged, so here it is!

Since it is my first contribution, I am sure I missed some things.

Bootterm is a tool to access serial devices with a simple user interface, and uncomplicated keyboard shortcuts.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/bo/bootterm/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  testVersion,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "bootterm";
+  version = "0.5";
+
+  src = fetchFromGitHub {
+    owner = "wtarreau";
+    repo = finalAttrs.pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/bo/bootterm/package.nix:14]:
```suggestion
    repo = "bootterm";
```

File: pkgs/by-name/bo/bootterm/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    hash = "sha256-AYpO2Xcd51B2qVUWoyI190BV0pIdA3HfuQJPzJ4yT/U=";
+  };
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  passthru.tests = {
+    version = testVersion {
+      package = finalAttrs.finalPackage;
+      command = "${finalAttrs.meta.mainProgram} -V";
+      version = finalAttrs.version;
```
Comment (Aleksanaa) [inline pkgs/by-name/bo/bootterm/package.nix:25]:
```suggestion
```

File: pkgs/by-name/bo/bootterm/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    longDescription = ''
+      BootTerm is a simple, reliable and powerful terminal designed to
+      ease connection to ephemeral serial ports as found on various SBCs,
+      and typically USB-based ones.
+    '';
+    homepage = "https://github.com/wtarreau/bootterm";
+    license = licenses.mit;
+    mainProgram = "bt";
+    maintainers = with maintainers; [ deadbaed ];
+    platforms = platforms.linux ++ platforms.darwin ++ platforms.freebsd;
```
Comment (Aleksanaa) [inline pkgs/by-name/bo/bootterm/package.nix:40]:
```suggestion
  meta = {
    description = "Simple, reliable and powerful terminal to ease connection to serial ports";
    longDescription = ''
      BootTerm is a simple, reliable and powerful terminal designed to
      ease connection to ephemeral serial ports as found on various SBCs,
      and typically USB-based ones.
    '';
    homepage = "https://github.com/wtarreau/bootterm";
    license = lib.licenses.mit;
    mainProgram = "bt";
    maintainers = with lib.maintainers; [ deadbaed ];
    platforms = lib.platforms.unix;
```

**Review by Aleksanaa — APPROVED** [review]

**Comment by deadbaed** [comment]
Thanks for the feedback! Ofborg complains that `testVersion` is not found.

After grepping `testVersion` in `pkgs/by-name`, I see packages importing `testers` and using `testers.testVersion` instead.

I'll update my package and wait for Ofborg again.

**Pull request merged**